backend/src/main/java/com/bakdata/conquery/models/index/MapInternToExternMapper.java

```diff
@@ -89,11 +115,23 @@ public String external(String internalValue) {
 			return internalValue;
 		}
 
-		return int2ext.getOrDefault(internalValue, internalValue);
+		if (int2ext.isCompletedExceptionally() || int2ext.isCancelled()) {
+			return internalValue;
+		}
+
+		try {
+			return int2ext.get().getOrDefault(internalValue, internalValue);
+		} catch (InterruptedException | ExecutionException e) {
+			// Should never be reached
+			log.warn("Unable to resolve mapping for internal value {} (enable TRACE for exception)", internalValue, (Exception) (log.isTraceEnabled() ? e : null));
+			return internalValue;
+		}
 	}
 
 	@Override
 	public InternToExternMapperId createId() {
 		return new InternToExternMapperId(getDataset().getId(), getName());
 	}
+
+	public static class Initializer extends Initializing.Converter<MapInternToExternMapper> {}
```

Comment:
Musst du den eigentlich überhaupt separat initialisieren? Oder kannst du nicht eine Klasse für alle verwenden?

Reply:
Ich werde es mal mit der parametrisierten Klasse probieren

Reply:
Das klappt leider nicht, da beschwert sich Jackson, dass er für das Interface `Initializing` (why?) keinen Creator findet, auch wenn ich die `Converter` in eine eigene Datei packe beschwert er sich weiter.

Ich lasse es so